src/core/thread/network_data_notifier.cpp

```diff
@@ -201,6 +204,10 @@ Error Notifier::SendServerDataNotification(uint16_t aOldRloc16, const NetworkDat
         tlv.SetLength(aNetworkData->GetLength());
         SuccessOrExit(error = message->Append(tlv));
         SuccessOrExit(error = message->AppendBytes(aNetworkData->GetBytes(), aNetworkData->GetLength()));
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BORDER_ROUTER_SIGNAL_NETWORK_DATA_FULL
+        Get<Leader>().CheckForNetDataGettingFull(*aNetworkData, aOldRloc16);
```

Comment:
Is it useful to send the Server Data Notification message if we've determined that the Network Data has no space?

Reply:
I think it may still be ok to try to register with leader. We may have an outdated view of leader's Network Data and there may be room to add our entries